set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java

```diff
@@ -0,0 +1,156 @@
+            writeElevation(getConfiguration().getElevationFileLocation(), xml);
+            // returning the elevation response
+            //TODO - remove the body, once we know how elevation file will be used
+            return new ResponseEntity<>(xml, HttpStatus.OK);
+        } else {
+            throw new UserSetNotFoundException(UserSetI18nConstants.ELEVATION_NOT_GENERATED,
+                    UserSetI18nConstants.ELEVATION_NOT_GENERATED,
+                    null);
+        }
+    } catch (IOException e) {
+        throw new InternalServerException(e);
+    }
+    }
+
+    /**
+     * Build elevation object
+     * @param userSets
+     * @return
+     */
+    private Elevation buildElevation(List<PersistentUserSet> userSets) throws UserSetNotFoundException {
+    if (userSets.isEmpty()) {
+        throw new UserSetNotFoundException(UserSetI18nConstants.ENTITY_USER_SET_NOT_FOUND,
+                    UserSetI18nConstants.ENTITY_USER_SET_NOT_FOUND,
+                    null);
+        }
+    Elevation elevation = new Elevation();
+    List<Query> queries = new ArrayList<>();
+    for (UserSet userset : userSets) {
+        String text = generateQueryText(userset.getSubject());
+        List<Doc> docList = new ArrayList<>();
+        if (userset.getItems() != null) {
+            for(String item : userset.getItems()) {
+            docList.add(new Doc(item));
+            }
+        }
+        if (!text.isEmpty() && !docList.isEmpty()) {
+            queries.add(new Query(text, docList));
+        }
+    }
+    if (!queries.isEmpty()) {
+        elevation.setQuery(queries);
+        return elevation;
+    }
+    return null;
+    }
+
+    /**
+     * get the entity reference from the subject list
+     * @param subject
+     * @return
+     */
+    private String generateQueryText(List<String> subject) {
+      StringBuilder text = new StringBuilder();
+      for(String s : subject) {
+          if (s.contains(WebUserSetFields.AGENT)) {
+              text.append(WebUserSetFields.ELEVATION_AGENT_QUERY);
+          }
+          else if (s.contains(WebUserSetFields.CONCEPT)) {
+              text.append(WebUserSetFields.ELEVATION_CONCEPT_QUERY);
+          }
+          else if(s.contains(WebUserSetFields.TIMESPAN)) {
+              text.append(WebUserSetFields.ELEVATION_TIMESPAN_QUERY);
+          } else {
+              break;
+          }
+          text.append("\"");
+          text.append(s);
+          text.append("\"");
+
+      }
+      return URLEncoder.encode(text.toString(), StandardCharsets.UTF_8);
+    }
+
+    /**
+     * writes the elevation in a file
+     *
+     * @param directoryLocation folder location
+     */
+    public static void writeElevation(String directoryLocation, String xml) throws InternalServerException {
+    if (!directoryLocation.isEmpty()) {
+        try (PrintWriter writer = new PrintWriter(directoryLocation + WebUserSetFields.SLASH + WebUserSetFields.ELEVATION_FILENAME, StandardCharsets.UTF_8)) {
```

Comment:
we should use here the org.apache.commons.io.FileUtils.write(File, string, encoding) and to call file.mkdirs() to ensure that the parent directories are created automatically on the server.

Reply:
done